R/feature.align.R

```diff
@@ -1,178 +1,194 @@
-feature.align <-
-function(features, min.exp=2,mz.tol=NA,chr.tol=NA,find.tol.max.d=1e-4, max.align.mz.diff=0.01, do.plot=TRUE, rt_colname="pos")     # returns a list of aligned features and original peak times
-{
-    if (do.plot) {
-        par(mfrow=c(3,2))
-        plot(c(-1,1),c(-1,1),type="n",xlab="",ylab="",main="",axes=FALSE)
-        text(x=0,y=0,"Feature alignment",cex=2)
-        plot(c(-1,1),c(-1,1),type="n",xlab="",ylab="",main="",axes=FALSE)
+to_attach <- function(pick, number_of_experiments, use = "sum") {
+    strengths <- rep(0, number_of_experiments)
+    if (is.null(nrow(pick))) {
+        strengths[pick[6]] <- pick[5]
+        return(c(pick[1], pick[2], pick[1], pick[1], strengths))
+    } else {
+        for (i in seq_along(strengths)) {
+            if (use == "sum")
+                strengths[i] <- sum(pick[pick[, 6] == i, 5])
+            if (use == "median")
+                strengths[i] <- median(pick[pick[, 6] == i, 5])
+        }
+        return(c(mean(pick[, 1]), mean(pick[, 2]), min(pick[, 1]),
+                 max(pick[, 1]), strengths))
     }
+}
 
-    to.attach<-function(this.pick, num.exp, use="sum")
-    {
-        this.strengths<-rep(0, num.exp)
-        if(is.null(nrow(this.pick)))
-        {
-            this.strengths[this.pick[6]]<-this.pick[5]
-            return(c(this.pick[1], this.pick[2], this.pick[1],this.pick[1],this.strengths))
-        }else{
-            for(m in seq_along(this.strengths))
-            {
-                if(use == "sum") this.strengths[m]<-sum(this.pick[this.pick[,6]==m,5])
-                if(use == "median") this.strengths[m]<-median(this.pick[this.pick[,6]==m,5])
-            }
-            return(c(mean(this.pick[,1]),mean(this.pick[,2]),min(this.pick[,1]),max(this.pick[,1]), this.strengths))
-        }
+# returns a list of aligned features and original peak times
+feature.align <- function(features,
+                          min_occurrence = 2,
+                          mz_tol_relative = NA,
+                          rt_tol_relative = NA,
+                          mz_max_diff = 1e-4,
+                          mz_tol_absolute = 0.01,
+                          do.plot = TRUE,
+                          rt_colname = "pos") {
+    if (do.plot) {
+        par(mfrow = c(3, 2))
+        plot(c(-1, 1), c(-1, 1), type = "n", xlab = "",
+             ylab = "", main = "", axes = FALSE)
+        text(x = 0, y = 0, "Feature alignment", cex = 2)
+        plot(c(-1, 1), c(-1, 1), type = "n", xlab = "",
+             ylab = "", main = "", axes = FALSE)
```

Comment:
Could be extracted to a `plot_xzy` function and moved to `plot.R`

Reply:
Extracted in [ee5a295](https://github.com/RECETOX/recetox-aplcms/pull/63/commits/ee5a2950c9de8fad8d302a8321fc5c21cb0ca213).